portfolio/src/main/webapp/script.js

```diff
@@ -56,3 +56,25 @@ function getRandomName() {
     document.getElementById('name-container').innerText = name;
   });
 }
+
+/**
+ * Fetches names from DataServlet and adds them to the DOM as a list.
+ */
+function getNames() {
+  fetch('/data').then(response => response.json()).then((names) => {
+
+    const namesListElement = document.getElementById('names-container');
+    namesListElement.innerHTML = '';
+    for (const nameIndex in names) {
+      console.log(names[nameIndex]);
+      namesListElement.appendChild(
+          createListElement('The ' + (+nameIndex + 1).toString() + '-th name is ' + names[nameIndex]));
```

Comment:
Is there one too many '+' in there?

```The` ' + (+nameIndex + 1).toString() +```

The second + looks a bit weird?

Reply:
The second + converts the nameIndex from String to Number. Is it better to use parseInt() instead?

Reply:
Perhaps using template listerals (per Libo's suggestion below) would make this easier to understand?
https://google.github.io/styleguide/jsguide.html#features-strings-template-strings

Reply:
Yep I agree - template strings are a good choice here. So you'd end up with something like 

`The ${nameIndex + 1}-th name is ${names[nameIndex]}`

(I can't do the ` backticks here without it getting markdown-formatted, but template literals are surrounded by backticks and not ' or ")

Generally speaking we really value code that is clear, easy to understand and readable, even if it means the code is a little bit longer/more verbose.  Hopefully you agree that this approach is pretty clear and easy to read without all of the string concatenation in the way! :)